src/Interpreters/Cache/FileSegment.h

```diff
@@ -205,6 +205,7 @@ friend class FileCache; /// Because of reserved_size in tryReserve().
 
     /// Write data into reserved space.
     void write(const char * from, size_t size, size_t offset);
+    void write(WriteBufferFromFile & wb, size_t offset);
```

Comment:
Correct if comment needed.

Also the meaning of `offset` is not clear, Is it the total number of bytes written before, with previous `write` calls?


```suggestion
    /// Write data into reserved space.
    void write(const char * from, size_t size, size_t offset);

    /** Use this overload if you need to handle file writing externally.
      * It writes data already stored in the `wb` buffer, ensuring the file segment's invariants.
      * The write buffer should refer to the same path that `FileSegment::getPath` returns.
      * For each file segment instance, only one overload of `write` can be used.
      */
    void write(WriteBufferFromFile & wb, size_t offset);
```